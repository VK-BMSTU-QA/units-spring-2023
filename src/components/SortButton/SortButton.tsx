import React, { FC } from 'react';

import type { SortBy } from '../../types';

import './SortButton.css';

type SortButtonProps = {
    currentSort: SortBy;
    onSortButtonClick: () => void;
};

export const SortButton: FC<SortButtonProps> = ({
    onSortButtonClick,
    currentSort,
}) => {
    return (
        <button
            className="sort-button"
            data-testid="sort-button"
            onClick={onSortButtonClick}
        >
            Сортировать {currentSort}
        </button>
    );
};
