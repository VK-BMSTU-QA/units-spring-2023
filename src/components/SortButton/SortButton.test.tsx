import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SortButton } from './SortButton';

afterEach(jest.clearAllMocks);

describe('Button render', () => {
    it('should contain default sort text', () => {
        const onSortButtonClick = jest.fn();
        const renderedButton = render(
            <SortButton
                currentSort="по умолчанию"
                onSortButtonClick={onSortButtonClick}
            />
        );

        expect(renderedButton.getByRole('button')).toHaveTextContent(
            'Сортировать по умолчанию'
        );
    });

    it('should be clicked once', () => {
        const onSortButtonClick = jest.fn();
        const renderedButton = render(
            <SortButton
                currentSort="по умолчанию"
                onSortButtonClick={onSortButtonClick}
            />
        );

        expect(onSortButtonClick).toHaveBeenCalledTimes(0);
        fireEvent.click(renderedButton.getByRole('button'));
        expect(onSortButtonClick).toHaveBeenCalledTimes(1);
    });

    it('should render correctly', () => {
        const onSortButtonClick = jest.fn();
        const renderedButton = render(
            <SortButton
                currentSort="по умолчанию"
                onSortButtonClick={onSortButtonClick}
            />
        );

        expect(renderedButton.asFragment()).toMatchSnapshot();
    });
});
