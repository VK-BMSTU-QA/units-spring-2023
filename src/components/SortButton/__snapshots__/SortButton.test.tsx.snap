// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render SortButton snapshot 1`] = `
<DocumentFragment>
  <button
    class="sort-button"
  >
    Сортировать по умолчанию
  </button>
</DocumentFragment>
`;

exports[`Button render SortButton snapshot and button text 1`] = `
<DocumentFragment>
  <button
    class="sort-button"
  >
    Сортировать по умолчанию
  </button>
</DocumentFragment>
`;

exports[`Button render should contain sort text 1`] = `
<DocumentFragment>
  <button
    class="sort-button"
  >
    Сортировать по умолчанию
  </button>
</DocumentFragment>
`;
